File: week7/py/good_segment.py
while True:
    try:
        n, s = map(int, input().split())
        arr = list(map(int, input().split()))

        l = 0
        acc_sum = 0
        res = 0

        for r in range(n):
            acc_sum += arr[r]
            while acc_sum > s:
                acc_sum -= arr[l]
                l += 1
            res += (r - l + 1)

        print(res)

    except EOFError:
        break

File: week7/cpp/good_segment.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    long long n;
    long long s;
    while (cin >> n >> s) {
        vector<long long> arr(n);
        for (int i = 0; i < n; ++i) {
            cin >> arr[i];
        }
        int l = 0;
        long long accSum = 0, res = 0;
        for (int r = 0; r < n; ++r) {
            accSum += arr[r];
            while (accSum > s) {
                accSum -= arr[l++];
            }
            res += (r - l + 1);
        }
        cout << res << endl;
    }
    return 0;
}